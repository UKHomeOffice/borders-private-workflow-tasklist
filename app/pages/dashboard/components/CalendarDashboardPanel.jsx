import React from 'react';
import {withRouter} from "react-router";

class CalendarDashboardPanel extends React.Component {


    calendar() {
        this.props.history.replace({
            pathname: "/calendar"
        });
    }
    render() {
        return <li className="__card column-one-third">
            <a href="#" onClick={this.calendar.bind(this)} className="card__body">
                <span className="bold-large">calendar</span>
            </a>
            <div className="card__footer">
                <span className="font-small">Operational calendar</span>
            </div>
        </li>
    }
}

export default withRouter(CalendarDashboardPanel);