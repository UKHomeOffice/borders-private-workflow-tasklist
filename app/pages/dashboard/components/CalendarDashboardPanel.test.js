import React from 'react';
import Enzyme from 'enzyme';
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import {CalendarDashboardPanel} from './CalendarDashboardPanel';
import { Router} from "react-router-dom";
import { createMemoryHistory } from 'history'

Enzyme.configure({ adapter: new Adapter() });

describe('Calendar Dashboard Panel', () => {
  const mockStore = configureStore();
  let store;
  const initialState = {
    'calendar-page': {}
  };
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders calendar dashboard panel', async() => {
    const props = {
      hasActiveShift: true
    };
    const wrapper = await mount(<CalendarDashboardPanel
      store={store}
      {...props}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('navigates to calendar page on click', async() => {
    const history = createMemoryHistory("/calendar");

    const props = {
      history: history,
      hasActiveShift: true
    };
    const wrapper = await mount(<Router history={history}><CalendarDashboardPanel
      store={store}
      {...props}
    /></Router>);

    const calendarPageLink = wrapper.find('#calendarPageLink');
    expect(calendarPageLink.exists()).toEqual(true);

    calendarPageLink.simulate('click');
    expect(props.history.location.pathname).toEqual('/calendar');
  });
});
