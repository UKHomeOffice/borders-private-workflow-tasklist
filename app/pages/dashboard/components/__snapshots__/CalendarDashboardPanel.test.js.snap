// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Dashboard Panel renders calendar dashboard panel 1`] = `
<CalendarDashboardPanel
  hasActiveShift={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <li
    className="__card column-one-third"
    id="calendarPanel"
  >
    <a
      className="card__body"
      href="#"
      id="calendarPageLink"
      onClick={[Function]}
    >
      <span
        className="bold-large"
      >
        calendar
      </span>
    </a>
    <div
      className="card__footer"
    >
      <span
        className="font-small"
      >
        Operational calendar
      </span>
    </div>
  </li>
</CalendarDashboardPanel>
`;
